include/oneapi/dpl/pstl/iterator_defs.h

```diff
@@ -31,23 +31,18 @@ namespace __internal
 // SFINAE with a non-iterator type by providing a default value.
 template <typename _IteratorTag, typename... _IteratorTypes>
 auto
-__is_iterator_of(int) -> decltype(
-    ::std::conjunction<::std::is_base_of<
-        _IteratorTag, typename ::std::iterator_traits<::std::decay_t<_IteratorTypes>>::iterator_category>...>{});
+__is_iterator_of(int) -> typename ::std::conjunction<::std::is_base_of<
+    _IteratorTag, typename ::std::iterator_traits<::std::decay_t<_IteratorTypes>>::iterator_category>...>::type;
 
 template <typename... _IteratorTypes>
 auto
 __is_iterator_of(...) -> ::std::false_type;
 
 template <typename... _IteratorTypes>
-struct __is_random_access_iterator : decltype(__is_iterator_of<::std::random_access_iterator_tag, _IteratorTypes...>(0))
-{
-};
+using __is_random_access_iterator = decltype(__is_iterator_of<::std::random_access_iterator_tag, _IteratorTypes...>(0));
 
 template <typename... _IteratorTypes>
-struct __is_forward_iterator : decltype(__is_iterator_of<::std::forward_iterator_tag, _IteratorTypes...>(0))
-{
-};
+using __is_forward_iterator = decltype(__is_iterator_of<::std::forward_iterator_tag, _IteratorTypes...>(0));
```

Comment:
Here `__is_forward_iterator` is template alias which is exactly the same as `decltype(__is_iterator_of<::std::forward_iterator_tag, _IteratorTypes...>(0))`